src/utils/tokenExplorerApi.js

```diff
@@ -111,18 +111,24 @@ const fetchBlockScoutAPIData = async (address) => {
   }
 };
 
-export const fetchNativeBalance = async (address) => {
-  try {
-    const url = `https://blockscout.com/xdai/mainnet/api?module=account&action=balance&address=${address}`;
-    const response = await fetch(url);
-    const json = await response.json();
-    if (!json.result || json.status === '0') {
-      const msg = json.message;
-      throw new Error(msg);
+export const fetchNativeBalance = async (address, daochain) => {
+  if (daochain === '0x1' || daochain === '0x4' || daochain === '0x2a') {
+    // eth chains not supported yet
+    // may need to do something different for matic too
+
+  } else {
+    try {
+      const url = `https://blockscout.com/xdai/mainnet/api?module=account&action=balance&address=${address}`;
+      const response = await fetch(url);
+      const json = await response.json();
+      if (!json.result || json.status === '0') {
+        const msg = json.message;
+        throw new Error(msg);
+      }
+      return json;
+    } catch (error) {
+      throw new Error(error);
```

Comment:
I could be wrong, but putting a throw error here will crash the app if anything gets caught. Console.error(error) is likely what you're after.  

Also, I haven't QA'd this yet, but depending on the use case, we usually want React to do something if the fetch doesn't work. Some sort of error UI.

Reply:
Whoops, I thought this was inside a useEffect. Looks like you'd be catching the error in the React component that calls this.